diff --git a/setup.py b/setup.py
index ccaf411..798961c 100755
--- a/setup.py
+++ b/setup.py
@@ -51,7 +51,7 @@ else:
 
 
 # Locate the re2 module
-_re2_prefixes = ['/usr', '/usr/local', '/opt/', '/opt/local', os.environ['HOME'] + '/.local']
+_re2_prefixes = ['re2', '/usr', '/usr/local', '/opt/', '/opt/local', os.environ['HOME'] + '/.local']
 
 re2_prefix = ''
 for a in _re2_prefixes:
@@ -81,7 +81,7 @@ def main():
             '-O3', '-march=native', '-DNDEBUG']
     # Older GCC version such as on CentOS 6 do not support C++11
     if not platform.python_compiler().startswith('GCC 4.4.7'):
-        extra_compile_args.append('-std=c++11')
+        extra_compile_args.extend(['-std=c++11', '-stdlib=libstdc++'])
     ext_modules = [
         Extension(
             're2',
@@ -105,7 +105,7 @@ def main():
                 'warn.unreachable': True,
             })
     setup(
-        name='re2',
+        name='re2_wheels',
         version='0.2.23',
         description='Python wrapper for Google\'s RE2 using Cython',
         long_description=get_long_description(),
